metasmoke.py

```diff
@@ -502,43 +586,30 @@ def func(url, *args, ignore_down=False, **kwargs):
             try:
                 response = method(GlobalVars.metasmoke_host + url, *args, **kwargs)
             except Exception:
-                GlobalVars.metasmoke_failures += 1
-                if GlobalVars.metasmoke_failures >= MAX_FAILURES and not GlobalVars.metasmoke_down:
-                    GlobalVars.metasmoke_down = True
-                    chatcommunicate.tell_rooms_with(
-                        'debug', '**Warning**: {}: {} latest connections to '
-                        'metasmoke have failed. Disabling metasmoke'.format(
-                            GlobalVars.location, GlobalVars.metasmoke_failures))
+                GlobalVars.MSStatus.failed()
+                if ignore_down:
+                    # Means that this is a status ping
+                    Metasmoke.AutoSwitch.to_off()
                 # No need to log here because it's re-raised
                 raise  # Maintain minimal difference to the original get/post methods
             else:
-                GlobalVars.metasmoke_failures -= 1
-                if GlobalVars.metasmoke_failures < 0:
-                    GlobalVars.metasmoke_failures = 0
+                GlobalVars.MSStatus.succeeded()
+                if ignore_down:
+                    # Means that this is a status ping
+                    Metasmoke.AutoSwitch.to_on()
 
             return response
         return func
 
     get = request_sender.__func__(requests.get)
     post = request_sender.__func__(requests.post)
 
-    @staticmethod
-    def reset_failure_count():  # For use in other places
-        GlobalVars.metasmoke_failures -= 1
-        if GlobalVars.metasmoke_failures < 0:
-            GlobalVars.metasmoke_failures = 0
-
     @staticmethod
     def send_status_ping_and_verify_scanning_if_active():
         in_standby_mode = GlobalVars.standby_mode or GlobalVars.no_se_activity_scan
         if not in_standby_mode:
             # This is the active instance, so should be scanning. If it's not scanning, then report or go to standby.
-
-            # We're not changing the scan stats, so shouldn't need a lock.
-            current_scan_time = GlobalVars.post_scan_time
-            current_posts_scanned = GlobalVars.num_posts_scanned
-            if current_scan_time == Metasmoke.last_ping_scan_time \
-                    and current_posts_scanned == Metasmoke.last_ping_posts_scanned:
+            if GlobalVars.PostScanStat.get_stat() == GlobalVars.PostScanStat.get_snap():
```

Comment:
I'm concerned that implementing this as `PostScanStat.get_snap()` and `PostScanStat.snap()` invites those methods to be used elsewhere than in this single piece of code, when their use anywhere else will disrupt their use here. If it was me, I'd keep it just as a separate variable stored within this class. Obviously, it still needs a lock, but that can be implemented within this class.